### PR TITLE
Update wien2k for version 16 with libcx and elpa support.

### PR DESCRIPTION
**Pull request by akesandgren**
Version 16 adds support for libxc and elpa, with a bunch of new
questions in the siteconfig script.

### CODE REVIEW

File: easybuild/easyblocks/w/wien2k.py [hunk 1]
```diff
@@ -194,6 +203,41 @@ def configure_step(self):
...
+                    })
+                else:
+                    qanda.update({
+                        'Do you want to use ELPA? (y,N):': 'n',
+                    })
+
+                qanda.update({
+                     'You need to KNOW details about your installed MPI, ELPA, and FFTW ) (y/n)': 'y',
+                     'Do you want to automatically search for FFTW installations? (Y,n):': 'n',
+                     'Please specify the ROOT-path of your FFTW installation (like /opt/fftw3/) or accept present choice (enter):' : fftw_root,
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: Line too long, please wrap?

File: easybuild/easyblocks/w/wien2k.py [hunk 2]
```diff
@@ -194,6 +203,41 @@ def configure_step(self):
...
+                        'use LIBXC (that you have installed before)? (y,N):': 'N',
+                    })
+
+                elpa_root = get_software_root('ELPA')
+                if elpa_root:
+                    elpa_root = elpa_root + "/"
+                    qanda.update({
+                        'Do you want to use ELPA? (y,N):': 'y',
+                        'Do you want to automatically search for ELPA installations? (Y,n):': 'n',
+                        'Please specify the ROOT-path of your ELPA installation (like /usr/local/elpa/) or accept present path (Enter):': elpa_root,
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: Line too long, please wrap?

File: easybuild/easyblocks/w/wien2k.py [hunk 3]
```diff
@@ -127,9 +127,11 @@ def configure_step(self):
 
         rplibs = ' '.join(rplibs)
 
+	# Haven't figured out how to set the FOPT value correctly in the
+	# cfgscript yet.
         d = {
-             'FC': '%s %s' % (os.getenv('F90'), os.getenv('FFLAGS')),
-             'MPF': "%s %s" % (os.getenv('MPIF90'), os.getenv('FFLAGS')),
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: Why stop specifying the compiler flags @akesandgren?

File: easybuild/easyblocks/w/wien2k.py [hunk 4]
```diff
@@ -127,9 +127,11 @@ def configure_step(self):
 
         rplibs = ' '.join(rplibs)
 
+	# Haven't figured out how to set the FOPT value correctly in the
+	# cfgscript yet.
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: Is this something that should be fixed first?

File: easybuild/easyblocks/w/wien2k.py [hunk 5]
```diff
@@ -150,6 +152,12 @@ def configure_step(self):
             line = re.sub('(\s+)exit 1', '\\1exit 0', line)
             sys.stdout.write(line)
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: we should be using `apply_regex_substitutions` here too...

File: easybuild/easyblocks/w/wien2k.py [hunk 6]
```diff
@@ -194,6 +203,41 @@ def configure_step(self):
...
             })
+
+            if LooseVersion(self.version) >= LooseVersion("16"):
+                libxc_root = get_software_root('libxc')
+                if libxc_root:
+                    libxc_root = libxc_root + "/"
+                    qanda.update({
+                        'use LIBXC (that you have installed before)? (y,N):': 'y',
+                        'Do you want to automatically search for LIBXC installations? (Y,n):': 'n',
+                        'Please specify the ROOT-path of your LIBXC installation (like /usr/local/libxc/) or accept the present path (Enter) (EXIT or an empty LIBXCpath aborts LIBXC selection):': libxc_root,
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: Line too long, please wrap.

File: easybuild/easyblocks/w/wien2k.py [hunk 7]
```diff
@@ -194,6 +203,41 @@ def configure_step(self):
                  'Please specify the ROOT-path of your FFTW installation (like /opt/fftw3):' : fftw_root,
                  'is this correct? enter Y (default) or n:' : 'Y',
             })
+
+            if LooseVersion(self.version) >= LooseVersion("16"):
+                libxc_root = get_software_root('libxc')
+                if libxc_root:
+                    libxc_root = libxc_root + "/"
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: this is really needed? :-/

File: easybuild/easyblocks/w/wien2k.py [hunk 8]
```diff
@@ -194,6 +203,41 @@ def configure_step(self):
...
+                    qanda.update({
+                        'Do you want to use ELPA? (y,N):': 'n',
+                    })
+
+                qanda.update({
+                     'You need to KNOW details about your installed MPI, ELPA, and FFTW ) (y/n)': 'y',
+                     'Do you want to automatically search for FFTW installations? (Y,n):': 'n',
+                     'Please specify the ROOT-path of your FFTW installation (like /opt/fftw3/) or accept present choice (enter):' : fftw_root,
+                     'Is this correct? enter Y (default) or n:' : 'Y',
+                     'Is this correct? Enter Y (default) or n:' : 'Y',
```
Comment (boegel) [inline easybuild/easyblocks/w/wien2k.py]: you can make this a stdqa entry which supports regex's?

**Comment by akesandgren** [comment]
Maybe...
We can close this one and i can take a look at #1405 and see if there are any important differences.

Will take a while before i have time to do that though.

**Comment by migueldiascosta** [comment]
I had missed this one, would have saved me some time :p 

there are some differences between versions 16 and 17 regarding `libxc` (different messages) and `ELPA` (selection of version), the merged block still needs to be tweaked if 16 is to be supported...

**Pull request closed without merging**